### PR TITLE
init: v0 of Krypton UI

### PR DESCRIPTION
**Pull request by shraddhaa98**
This is the initial version of Krypton UI. Would appreciate a code review on this!

### CODE REVIEW

File: components/carousel.css [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+.carousel__button--hidden {
+    display: none;
+}
+
+.carousel__nav {
+    display: flex;
+    justify-content: center;
+    padding: 30px 0;
```
Comment (Joyan11) [inline components/carousel.css]: since rem is used everywhere. use the same unit here for better consistency

File: components/carousel.css [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+.carousel__dot {
+    width: 1rem;
+    height: 1rem;
+    border-radius: 50%;
+    border: 0;
+    background: rgba(0, 0, 0, 0.3);
+    margin: 0 12px;
```
Comment (Joyan11) [inline components/carousel.css]: a similar suggestion on L75

**Review by Joyan11 — COMMENTED** [review]
Hello,
Impressive work with the classes. nice to see BEM architecture being used.
Pseudo classes are used very well as well as keyframes. the resize and appearance are some new properties i saw. 
some minor issues i found with the UI was that the Navbar is going outside the viewport and the home page isn't completely visible on desktop. This is probably because of the viewport height. I've mentioned some changes below regarding units, keep it uniform especially for padding and margin.
Overall Great work with the Code . Cheers!

**Comment by shilpa3495** [comment]
Hi @shraddha319 
Everything looks fine, these are some of my suggestions,

1. Add a favicon to your website.
2. The sidebar affects the main content due to the sidebar `position: fixed`
3. Change the colour of the alert cross icon similar to toast.
4. Components sub nav list decrease padding-left. You can take the reference of MDN.
5. You can also add the transitions on hover for smoothness for the user.
6. Padding is uneven in the modal footer section.

File: components/carousel.css [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+.carousel__image {
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+}
+
+.carousel__track-container {
+    position: relative;
```
Comment (shilpa3495) [inline components/carousel.css:14]: you don't need to define multiple times of same property you can write it like that:-
```
.carousel, .carsousel__track-container{
position:relative;
}
```
and similar for other properties that is used multiple times.

**Review by shilpa3495 — COMMENTED** [review]
Hi @shraddha319
Everything looks fine, these are some of my suggestions,

- Add a favicon to your website.
- The sidebar affects the main content due to the sidebar position: fixed
- Change the colour of the alert cross icon similar to toast.
- Components sub nav list decrease padding-left. You can take the reference of MDN.
- You can also add the transitions on hover for smoothness for the user.
- Padding is uneven in the modal footer section.

File: components/card.css [hunk 4]
```diff
@@ -0,0 +1,161 @@
+.card {
+    min-width: 400px;
+    max-width: 700px;
```
Comment (theniteshnarang) [inline components/card.css:2]: I don't get why you had to put min and max-width, as it's showing no effect if I remove those.

File: components/card.css [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
+
+.card--text-overlay .card__body {
+    color: var(--clr-text);
+    min-width: 100%;
+    min-height: 70%;
+    max-width: 80%;
```
Comment (theniteshnarang) [inline components/card.css]: There's no need to write ```max-width``` here, as it's always over-ridden by the ```min-width```.

Read more for your ref: [CSS tricks link](https://css-tricks.com/almanac/properties/m/max-width/#:~:text=The%20max%2Dwidth%20property%20in,after%20width%20in%20your%20declaration.)

File: components/toast.css [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    position: fixed;
+    /* top: 10px;
+    right: 10px; */
+    bottom: 50px;
+    left: 50%;
+    transform: translateX(-50%);
+    font-size: 0.9rem;
+    font-family: inherit;
+    width: 480px;
+    min-width: 400px;
```
Comment (theniteshnarang) [inline components/toast.css]:
```suggestion
    
```
It can be removed.

**Review by theniteshnarang — CHANGES_REQUESTED** [review]
Hey @shraddha319,
I loved your work so much, so consistent with naming and layouts, and excellent UI.

Your library is the one that one should look for if they want to follow the right practices for doing things. Great source of learning!

**Comment by shraddhaa98** [comment]
> Hello,
> Impressive work with the classes. nice to see BEM architecture being used.
> Pseudo classes are used very well as well as keyframes. the resize and appearance are some new properties i saw.
> some minor issues i found with the UI was that the Navbar is going outside the viewport and the home page isn't completely visible on desktop. This is probably because of the viewport height. I've mentioned some changes below regarding units, keep it uniform especially for padding and margin.
> Overall Great work with the Code . Cheers!

Hi Joyan! I have updated the units and tried keeping them consistent where ever possible, thanks for pointing this out. Also, I have updated the home page layout.  

Thank you for reviewing my code. :)

File: components/carousel.css (hunk 1 above)
Comment (shraddhaa98) [reply components/carousel.css]: updated it

File: components/carousel.css (hunk 2 above)
Comment (shraddhaa98) [reply components/carousel.css]: updated it

File: components/carousel.css (hunk 3 above)
Comment (shraddhaa98) [reply components/carousel.css:14]: updated it

**Comment by shraddhaa98** [comment]
> Hi @shraddha319
> Everything looks fine, these are some of my suggestions,
> 
> * Add a favicon to your website.
> * The sidebar affects the main content due to the sidebar position: fixed
> * Change the colour of the alert cross icon similar to toast.
> * Components sub nav list decrease padding-left. You can take the reference of MDN.
> * You can also add the transitions on hover for smoothness for the user.
> * Padding is uneven in the modal footer section.

Hi Shilpa! I have made all the changes that you suggested. 

Thank you for reviewing my code :)

File: components/card.css (hunk 4 above)
Comment (shraddhaa98) [reply components/card.css:2]: Fixed it

File: components/card.css (hunk 5 above)
Comment (shraddhaa98) [reply components/card.css]: Fixed it

File: components/toast.css (hunk 6 above)
Comment (shraddhaa98) [reply components/toast.css]: Fixed it

**Comment by shraddhaa98** [comment]
> Hey @shraddha319,
> I loved your work so much, so consistent with naming and layouts, and excellent UI.
> 
> Your library is the one that one should look for if they want to follow the right practices for doing things. Great source of learning!

Hi Nitesh!
Thank you so much for the kind words :)

I have made all the changes you suggested. Regarding the width property, my intension was to set a max and min value range for the width property but I did not know one of them was being overridden. Thank you for pointing this out. The link you provided was very helpful!

**Pull request merged**